R/ABSClass.R

```diff
@@ -68,51 +81,75 @@ Zhu23ABS <- R6::R6Class(
-    #' @param start_point The start point of each trial.
+    #' @param width a numeric value of the standard deviation of the proposal distribution for MC3.
+    #' @param n_chains an integer of the number of chains for the sampler.
+    #' @param nd_time a numeric value of the non-decision time (in seconds). When `s_nd_time` is not 0, `nd_time` represents the lower bound of the non-decision time.
+    #' @param s_nd_time a numeric value of the inter-trial-variability of the non-decition time (in seconds).
+    #' @param lambda a numeric value of the rate parameter of the Erlang distribution for decision time.
+    #' @param distr_name a character string indicating the type of the posterior hypothesis distribution.
+    #' @param distr_params a numeric vector of the additional parameters for the posterior hypothesis distribution.
+    #' @param custom_density a function that returns a distribution when the user prefer a customed posterior hypothesis distribution.
     #' 
     #' @return A new 'Zhu23ABS' object.
     #'
     #' @examples
-    #' zhuabs <- Zhu23ABS$new(nd_time = 0.3, s_nd_time = 0.5, lambda = 10, n_chains = 5, width = 1, distr_name = 'norm')
+    #' zhuabs <- Zhu23ABS$new(width = 1, n_chains = 5, nd_time = 0.3, s_nd_time = 0.5, lambda = 10)
     #' 
-    initialize = function(nd_time, s_nd_time=0, lambda, n_chains, width, distr_name='norm', start_point=NA) {
-      super$initialize(n_chains, distr_name, start_point, nd_time, s_nd_time)
+    initialize = function(width, n_chains, nd_time, s_nd_time, lambda, distr_name = 'norm', distr_params = 1, custom_density = NULL) {
+      super$initialize(n_chains, nd_time, s_nd_time, distr_name, distr_params, custom_density)
       
-      stopifnot("lambda should be a single numeric value."=(is.numeric(lambda) && length(lambda) == 1))
-      stopifnot("width should be a single numeric value."=(is.numeric(width) && length(width) == 1))
+      stopifnot('Argument "lambda" should be a single numeric value.'=(is.numeric(lambda) && length(lambda) == 1))
+      stopifnot('Argument "width" should be a single numeric value.'=(is.numeric(width) && length(width) == 1))
       
       self$lambda <- lambda
       self$width <- width
     },
     
-    
     #' @description
-    #' This function is for simulating two-alternative-force choice tasks by Zhu23ABS.
+    #' Simulate the ABS model.
+    #' 
+    #' 
+    #' @param stopping_rule a character string indicating the stopping rule of ABS to be applied. Possible values are `"fixed"` and `"relative"`. See also `Details`.
+    #' @param start_point a numeric vector setting the start point of each trial for the sampler. By default, it's set to `NA`, indicating that the starting point of the first trial is a random point from the posterior of hypotheses, and the starting points of subsequent trials are set to the last sample of the previous trial. For more detailed information, please refer to the vignette.
```

Comment:
maybe name the vignette? As in, "please refer to the 'Simulations of ABS' vignette"